# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/ipc/PWebGPU.ipdl
# Commit: dfd34fb1aade
# Full Hash: dfd34fb1aade1295579715b1c21f1e51bbfe370e
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-05-18 21:36:28
# Regressor Bug: 1709462
# File Overlap Count: 2
# Description:
#   Bug 1709462 - Fix freeing of WebGPU bind group layouts r=jimb
#   
#   When the pipeline layout is implicit at pipeline creation,
#   we collect the IDs of BGLs, to be able to produce GPUBindGroupLayout
#   object upon user request. However, the produced object didn't know if
# ==============================================================================

diff -r 5c6f0950714d -r dfd34fb1aade dom/webgpu/ipc/PWebGPU.ipdl
--- a/dom/webgpu/ipc/PWebGPU.ipdl	Tue May 18 16:24:49 2021 +0000
+++ b/dom/webgpu/ipc/PWebGPU.ipdl	Tue May 18 16:26:27 2021 +0000
@@ -59,6 +59,7 @@
   async ShaderModuleDestroy(RawId selfId);
   async ComputePipelineDestroy(RawId selfId);
   async RenderPipelineDestroy(RawId selfId);
+  async ImplicitLayoutDestroy(RawId implicitPlId, RawId[] implicitBglIds);
   async DeviceCreateSwapChain(RawId selfId, RawId queueId, RGBDescriptor desc, RawId[] bufferIds, ExternalImageId externalId);
   async SwapChainPresent(ExternalImageId externalId, RawId textureId, RawId commandEncoderId);
   async SwapChainDestroy(ExternalImageId externalId);